### PR TITLE
First steps to make a utils library

### PR DESCRIPTION
**Pull request by noelmarkham**
Restructuring the project into three (four actually) distinct projects:
  * `scalacheck-datetime` - the one we know and love
  * `magic` - inclusion of generators for "magic" strings (`drop database STUDENTS` etc)
  * `combinators` - generators for common use-cases (ie, a map and a list of elements guaranteed to be in the map)
 * `docs` - the fourth project, which is for the Jekyll site

`magic` and `combinators` are still in a skeleton form for this PR.

Also todo following this PR is the renaming in Github, docs and elsewhere to reflect use as a utils library rather than just datetime.

@raulraja would you be able to take a look please?

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -2,10 +2,10 @@ import com.typesafe.sbt.SbtGhPages.ghpages
 import com.typesafe.sbt.SbtGit.git
 import de.heikoseeberger.sbtheader.HeaderPattern
 
-lazy val buildSettings = Seq(
+lazy val commonSettings = Seq(
   organization := "com.fortysevendeg",
```
Comment (raulraja) [inline build.sbt]: Starting now and going forward we are gonna use `com.47deg` as the org name and groupId for all libs going forward. Existing libs will be migrated over as next releases happen to bring all the OS 47 libs under the same groupId. Once you are ready to publish to sonatype and central, let @fedefernandez and I know and we'll let you when ready. We need to request the org with the first lib that makes there. /cc @andyscott

File: magic/src/main/scala/com/fortysevendeg/scalacheck/magic/Magic.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import org.scalacheck.Gen._
+import org.scalacheck.Arbitrary.arbitrary
+
+object Magic {
+
+  private[magic] val strings = List("This", "is", "a", "temporary", "list", "of", "strings")
+
+  val magicStrings: Gen[String] = oneOf(strings)
+
+  implicit val enhancedStringArbitrary: Arbitrary[String] = Arbitrary {
```
Comment (raulraja) [inline magic/src/main/scala/com/fortysevendeg/scalacheck/magic/Magic.scala]: I wonder if for cases like this one where we are explicitly overriding the scalacheck default instance for `Arbitrary[String]` an alternative approach would be to use tagged types so it's explicit that we are summoning the implicit for magic Strings. `Arbitrary[String @@ Magic]`

File: build.sbt (hunk 1 above)
Comment (noelmarkham) [inline build.sbt]: I'll leave this in place here and add a ticket to address this before any future releases

File: magic/src/main/scala/com/fortysevendeg/scalacheck/magic/Magic.scala (hunk 2 above)
Comment (noelmarkham) [inline magic/src/main/scala/com/fortysevendeg/scalacheck/magic/Magic.scala]: I'll have a think about this. In my head the reasoning is you want arbitrary `String`s, and just have the added bonus benefit of checking some magic ones too. I see having an import to override the scalacheck default is essentially the same as tagging.

**Comment by noelmarkham** [comment]
Thanks!

Re the microsite changes, I'll do this in a separate PR before a release.

**Pull request merged**